### ISSUE TITLE
header of output different from input 

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#596]
data: mt from batch_processing
branch: master

~~~
fslhd mt0mt1.nii.gz
...
qto_xyz:1      -0.843379  -0.000000  -0.503756  78.196503
qto_xyz:2      -0.001874  0.841376  1.273740  -80.194565
qto_xyz:3      -0.024932  -0.063247  16.944729  -18.327135
qto_xyz:4      0.000000  0.000000  0.000000  1.000000
qform_xorient  Right-to-Left
qform_yorient  Posterior-to-Anterior
qform_zorient  Inferior-to-Superior
sform_name     Scanner Anat
sform_code     1
sto_xyz:1      -0.843379  -0.000000  -0.503756  78.196503
sto_xyz:2      -0.001874  0.841376  1.273740  -80.194565
sto_xyz:3      -0.024932  -0.063247  16.944727  -18.327135
sto_xyz:4      0.000000  0.000000  0.000000  1.000000
~~~
versus
~~~
fslhd mt0mt1_wmseg.nii.gz
...
qto_xyz:1      -0.843379  -0.000000  -0.503756  98.306412
qto_xyz:2      -0.001874  0.841376  1.273741  -100.212029
qto_xyz:3      -0.024932  -0.063247  16.944729  -16.224560
qto_xyz:4      0.000000  0.000000  0.000000  1.000000
qform_xorient  Right-to-Left
qform_yorient  Posterior-to-Anterior
qform_zorient  Inferior-to-Superior
sform_name     Scanner Anat
sform_code     1
sto_xyz:1      -0.299868  -0.000000  -0.503756  98.306412
sto_xyz:2      -0.000666  0.299156  1.273741  -100.212029
sto_xyz:3      -0.008865  -0.022488  16.944727  -16.224560
sto_xyz:4      0.000000  0.000000  0.000000  1.000000
~~~

**Comment by SaraDupont** [comment]
This is due to the padding/cropping I add to fix issue #587. I'm working on it.

**Comment by SaraDupont** [comment]
The difference between mt0mt1 and mt0mt1_wmseg headers: 
  -  in the diagonal of the sform matrix comes from the resampling : ```sct_resample``` seems to update only the qform matrix
  - in the last column in the right (image origin) comes from the padding

**Comment by SaraDupont** [comment]
Copying the original target header to the results works. I'm looking for a cleaner solution but this will address the issue.

**Comment by SaraDupont** [comment]
Fixed for now, but we shouldn't need to copy the header : it should be correct !

**Comment by SaraDupont** [comment]
In sct_resample, the sform wasn't correctly set (the qform was OK) : this was fixed in commit b3f4a2e.

**Comment by jcohenadad** [comment]
great :+1:
